src/index.js

```diff
@@ -39,8 +41,7 @@ exports.unmarshalPublicKey = (buf) => {
 exports.marshalPublicKey = (key, type) => {
   type = (type || 'rsa').toLowerCase()
 
-  // for now only rsa is supported
-  if (type !== 'rsa') {
+  if (type !== 'rsa' && type !== 'ed25519') {
```

Comment:
Sweet, just pushed this up.  I used `if (KEY_TYPES.indexOf(type) < 0)` for this, since I can't remember when `Array.includes()` became a thing or whether you're using any polyfills for this lib.  😄